python/ray/autoscaler/v2/instance_manager/reconciler.py

```diff
@@ -760,15 +782,12 @@ def _handle_ray_status_transition(
                 )
 
         for cloud_instance_id, ray_node in ray_nodes_by_cloud_instance_id.items():
-            if cloud_instance_id not in im_instances_by_cloud_instance_id:
-                # This is a ray node that's not managed by the instance manager.
-                # or we haven't discovered the instance yet. There's nothing
-                # much we could do here.
-                logger.info(
-                    f"Ray node {binary_to_hex(ray_node.node_id)} has no matching "
-                    f"instance with cloud instance id={cloud_instance_id}."
-                )
-                continue
+            assert cloud_instance_id in im_instances_by_cloud_instance_id, (
```

Comment:
For out-of-band ray node, we do nothing?

Reply:
we will upsert out-of-band ray node's cloud instance with a new ALLOCATED IM instance just like how we handle out-of-band cloud instance. And let the reconcilation handle the rest.